web-js/src/components/useTodos_graphql.js

```diff
@@ -0,0 +1,164 @@
+import React from "react";
+import { ApolloClient, gql, HttpLink, InMemoryCache } from "@apollo/client";
+import { APP_ID } from "./App";
+import { useRealmApp } from "./RealmApp";
+
+function useApolloClient() {
+  const realmApp = useRealmApp();
+  if (!realmApp.currentUser) {
+    throw new Error(`You must be logged in to Realm to call useApolloClient()`);
+  }
+
+  const client = React.useMemo(() => {
+    const graphqlUri = `https://realm.mongodb.com/api/client/v2.0/app/${APP_ID}/graphql`;
+    // Local apps should use a local URI!
+    // const graphqlUri = `https://us-east-1.aws.stitch.mongodb.com/api/client/v2.0/app/${APP_ID}/graphql`
+    async function getValidAccessToken() {
+      // An already logged in user's access token might be stale. To guarantee that the token is
+      // valid, we refresh the user's custom data which also refreshes their access token.
+      await realmApp.currentUser.refreshCustomData();
+      return realmApp.currentUser.accessToken
+    }
+    return new ApolloClient({
+      link: new HttpLink({
+        uri: graphqlUri,
+        // We define a custom fetch handler for the Apollo client that lets us authenticate GraphQL requests.
+        // The function intercepts every Apollo HTTP request and adds an Authorization header with a valid
+        // access token before sending the request.
+        fetch: async (uri, options) => {
+          const accessToken = await getValidAccessToken();
+          options.headers.Authorization = `Bearer ${accessToken}`;
+          return fetch(uri, options);
+        },
+      }),
+      cache: new InMemoryCache(),
+    });
+  }, [realmApp.currentUser]);
+
+  return client;
+}
+
+export function useTodos() {
+  const realmApp = useRealmApp();
+  const graphql = useApolloClient();
+  const taskCollection = React.useMemo(() => {
+    const mdb = realmApp.currentUser.mongoClient("mongodb-atlas");
+    return mdb.db("todo").collection("Task");
+  }, [realmApp.currentUser]);
+  const [todos, setTodos] = React.useState([]);
+  
+  const [loading, setLoading] = React.useState(true);
+  React.useEffect(() => {
+    const fetchTodos = async () => {
+      const { data } = await graphql.query({
+        query: gql`
+          query FetchAllTasks {
+            tasks {
+              _id
+              _partition
+              isComplete
+              summary
+            }
+          }
+        `,
+      });
+      return data.tasks
+    };
+
+    const watchTodos = async () => {
```

Comment:
Can you add more comments to this method? Explaining what the loops are doing, what the anonymous functions are doing, and what is being returned.